assembly/std/arguments.ts

```diff
@@ -0,0 +1,212 @@
+import {Address} from './address';
+import {ByteArray} from '@massalabs/as/assembly/byteArray';
+
+/**
+ * Args for remote function call.
+ *
+ * This class can serialize assembly script native types into bytes, in order to
+ * make smart-contract function call easier.
+ *
+ * In a smart-contract exposed function, use this class to deserialize the string
+ * argument, using the `next...` methods.
+ *
+ * In a smart-contract, to call another smart-contract function, use this class
+ * to serialize the arguments you want to pass to the smart-contract function
+ * call.
+ *
+ */
+export class Args {
+  private offset: i64 = 0;
+  private serialized: string = '';
+
+  /**
+   *
+   * @param {string} serialized
+   */
+  constructor(serialized: string = '') {
+    this.serialized = serialized;
+  }
+
+  /**
+   * Returns the serialized string to pass to CallSC.
+   *
+   * @return {string} the serialized string
+   */
+  serialize(): string {
+    return this.serialized;
+  }
+
+  // getters
+
+  /**
+   * Returns the deserialized address.
+   *
+   * @return {Address} the address
+   */
+  nextAddress(): Address {
+    const address = new Address();
+    this.offset = address.fromStringSegment(
+      this.serialized,
+      this.offset as i32,
+    );
+    return address;
+  }
+
+  /**
+   * Returns the deserialized string.
+   *
+   * @return {string} the string
+   */
+  nextString(): string {
+    const length = this.nextU32();
+    let offset: i32 = this.offset as i32;
+    const end = offset + length;
+    const result = this.serialized.slice(offset, end);
+    this.offset = end;
+    return result;
+  }
+
+  /**
+   * Returns the deserialized number as u64.
+   *
+   * @return {u64}
+   */
+  nextU64(): u64 {
+    const byteArray = this.fromByteString(this.serialized);
+    const amount = this.toU64(byteArray, this.offset as u8);
```

Comment:
amout is a bad copy/paste I think.
```suggestion
    const value = this.toU64(byteArray, this.offset as u8);
```

Reply:
indeed, I have renamed all `amount` into `value`